pinot-core/src/main/java/com/linkedin/pinot/core/realtime/impl/kafka/MessageBatch.java

```diff
@@ -16,13 +16,14 @@
 
 package com.linkedin.pinot.core.realtime.impl.kafka;
 
-import kafka.javaapi.consumer.SimpleConsumer;
-
-
 /**
- * Factory for Kafka SimpleConsumer instances, so that we can create mock instances of the Kafka SimpleConsumer for
- * unit tests.
+ * Interface wrapping likafka consumer.
+ * @param <T>
  */
-public interface KafkaSimpleConsumerFactory {
-  SimpleConsumer buildSimpleConsumer(String host, int port, int soTimeout, int bufferSize, String clientId);
-}
+public interface MessageBatch<T> {
+    int getMessageCount();
+    T getMessageAtIndex(int index);
+    int getMessageOffsetAtIndex(int index);
+    int getMessageLengthAtIndex(int index);
+    long getNextKafkaMessageOffsetAtIndex(int index);
```

Comment:
what does 'getNextKafkaMessageOffsetAtIndex()' do? I am guessing it gets the offset of the next message? What if this is the last message in the batch?

Reply:
Yes, I've duplicated our current logic. currentOffset == finalOffset is checked inside endCriteriaReached.